### PR TITLE
🪟 🎨 Credits Usage Table Redesign pt. 1 (Migrated from `airbytehq/airbyte`)

### PR DESCRIPTION
**Pull request by tealjulia**
## What
Migration from https://github.com/airbytehq/airbyte/pull/23031

Original PR Description:

closes https://github.com/airbytehq/airbyte-platform-internal/issues/4272

[Figma](https://www.figma.com/proto/568DVwWXTcagwhoB2r1Qke/01_01_BILLING?page-id=714%3A7293&node-id=714%3A8936&viewport=194%2C464%2C0.24&scaling=min-zoom&starting-point-node-id=714%3A8936)

Addressed:
- Adds a feature flag to use going forward with the redesign.
- Adds additional columns to the Credits Usage Table on the credits page. (NOTE: Schedule data is being added to the API, but I added a static column for it to get the styling right for now)
- Also renames it to Billing page.
- Links to connection/source/destination from table

Not Addressed:
- Adding tooltips to ellipsed cells (waiting on https://github.com/airbytehq/airbyte-platform/pull/112 will fix in later PR)

Screenshot 2023-02-16 at 3 55 20 PM

## Recommended reading order
1. `UsagePerConnectionTable.*`
2. others

## Can this PR be safely reverted / rolled back?
*If you know that your PR is backwards-compatible and can be simply reverted or rolled back, check the YES box.*

*Otherwise if your PR has a breaking change, like a database migration for example, check the NO box.*

*If unsure, leave it blank.*
- [X] YES 💚
- [ ] NO ❌

## 🚨 User Impact 🚨
This is the first part of a redesign for the Credits page.
Nearly all changes are behind the feature flag `billingPage.billingInsights` on Launch Darkly which is currently not targeting anyone in production.

### CODE REVIEW

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (tealjulia) [inline airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:136]: TODO: replace these with the new internal NavLink once merged.  I'm guessing they have hover styling baked in?  That was my main reason for not fiddling with that here right now!

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (josephkmh) [inline airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:42]: I know not related to the changes in this PR, but maybe we could type the expected query params for this route and let them live in `BillingPage.tsx` (as e.g. `BillingPageQueryParams`). That way if we add/remove more query params in this route, we can re-use the same type? Not at all blocking, just occurred to me while reading.

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (josephkmh) [inline airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:214]: Will schedule actually be sortable? Probably not super easy with cron expressions. If not, could we just make this a non-interactive table header?

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (josephkmh) [inline airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:221]:
```suggestion
                  formatMessage({id: "credits.unknownColumnValue" })
```

Where `credits.unknownColumnValue: "Not available"` or something similar. Just to avoid confusion if anyone is testing this, didn't read the PR description, and wonders why their connection schedules are all the same 🙂

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (josephkmh) [inline airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:201]:
```suggestion
                </FlexContainer>
```

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (josephkmh) [inline airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:167]:
```suggestion
                  </Text>
```

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (josephkmh) [reply airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:136]: > I'm guessing they have hover styling baked in?

At the moment, no 🙁. Waiting for design feedback before adding anything. IMO we could go ahead and use the non-hover-styled `<Link>` now (it's been merged) and once styling is added it will just work™.

Any reason you're using `<NavLink>` here instead of just `<Link>`?

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (josephkmh) [inline airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:259]: Could we combine this column with the bar chart? I think semantically they belong together?

File: airbyte-webapp/src/hooks/services/Experiment/experiments.ts
Comment (josephkmh) [inline airbyte-webapp/src/hooks/services/Experiment/experiments.ts:29]: nitpick:
```suggestion
  "billing.billingInsights": boolean;
```

I know `authPage` follows a "page-first" approach, but I feel like targeting experiments by domain is more flexible. For example, if we want to toggle something in the sidebar or connection page related to this feature, it's not strictly on the `billingPage`.

File: airbyte-webapp/src/components/ui/Card/Card.tsx
Comment (josephkmh) [inline airbyte-webapp/src/components/ui/Card/Card.tsx:45]: What's the reason for having a card variant with and without a line under the title? Can we push back to have more consistency here, instead of adding two variants that are very similar?

**Review by josephkmh — CHANGES_REQUESTED** [review]
Glad to see this coming to life! Tested locally and left a few comments.

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (tealjulia) [reply airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:136]: Nope.  Will use Link.

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (tealjulia) [reply airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:214]: Good call.  I was going off the connections list table as a reference for this column and it looks non-interactive there as well.

File: airbyte-webapp/src/hooks/services/Experiment/experiments.ts
Comment (tealjulia) [reply airbyte-webapp/src/hooks/services/Experiment/experiments.ts:29]: Sure!  It looks like I may not have permissions to archive the flag that exists, though.

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (tealjulia) [reply airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx:210]: Sure thing.

I swapped it out for "PLACEHOLDER!!" (to make it even more obvious it's a placeholder, since there wouldn't be an "unknown value" here in the final product)

File: airbyte-webapp/src/components/ui/Card/Card.tsx
Comment (tealjulia) [reply airbyte-webapp/src/components/ui/Card/Card.tsx:45]: I like them so much more without the line, but reverted that and will take the convo to a channel if I don't hear back Monday. :)

File: airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx
Comment (tealjulia) [reply airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx]: Yep!  Though, I am not 100% sure we'll continue to use the query params when we switch over to the new return type from the API. 

I'm going to have to noodle on that.

**Comment by tealjulia** [comment]
migrated

**Pull request closed without merging**
